crates/freeze/src/datasets/transactions.rs

```diff
@@ -66,44 +73,55 @@ impl CollectByBlock for Transactions {
             .await?
             .ok_or(CollectError::CollectError("block not found".to_string()))?;
         let schema = query.schemas.get_schema(&Datatype::Transactions)?;
-        let receipt = if schema.has_column("gas_used") | schema.has_column("success") {
-            Some(source.get_tx_receipts_in_block(&block).await?)
-        } else {
-            None
-        };
-        Ok((block, receipt, query.exclude_failed))
+        let receipts: Vec<Option<_>> =
```

Comment:
if the receipts were collected after filtering, this would require collecting less data and would simply the filtering code

Reply:
that will be much more efficient, but it will require `get_tx_receipts_in_block` function to take additional `transactions: Vec<H160>` parameter if `eth_getBlockReceipts` fails because currently the function reuses `block.transactions` as fallback... :D I will refactor it in this PR.

Reply:
applied in [27181e4](https://github.com/paradigmxyz/cryo/pull/111/commits/27181e47c08e383b1e1aed70c474e840bcea9394)